common/util/src/test/java/org/eclipse/dataspaceconnector/common/configuration/ConfigurationFunctionsTest.java

```diff
@@ -0,0 +1,61 @@
+package org.eclipse.dataspaceconnector.common.configuration;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ConfigurationFunctionsTest {
+
+    private Map<String, Object> allProperties;
+
+    @BeforeEach
+    void setUp() {
+        allProperties = Map.of(
+                "edc.datasource.default.user", "test-user",
+                "edc.datasource.default.password", "test-pwd",
+                "edc.datasource.default.driverClassName", "org.company.default.Driver",
+                "edc.datasource.another.user", "test-user2",
+                "edc.datasource.another.password", "test-pwd2",
+                "edc.datasource.another.driverClassName", "org.company.another.Driver",
+                "edc.datasource.another.sub.property2", "bar");
+    }
+
+    @Test
+    void hierarchical() {
+        var hierarchy = ConfigurationFunctions.hierarchical(allProperties);
+        assertThat(hierarchy).hasSize(1);
+        assertThat(hierarchy).containsKey("edc");
+    }
+
+    @Test
+    void hierarchical_withRoot() {
+        var hierarchy = ConfigurationFunctions.hierarchical(allProperties, "edc.datasource");
+
+        assertThat(hierarchy).hasSize(2);
+        assertThat(hierarchy.get("default")).containsExactlyInAnyOrderEntriesOf(Map.of(
+                "user", "test-user",
+                "password", "test-pwd",
+                "driverClassName", "org.company.default.Driver"));
+
+        assertThat(hierarchy.get("another")).containsExactlyInAnyOrderEntriesOf(Map.of(
+                "user", "test-user2",
+                "password", "test-pwd2",
+                "sub.property2", "bar",
+                "driverClassName", "org.company.another.Driver"));
+    }
+
+    @Test
+    void hierarchical_withRoot_deeperLevel() {
+
+        var hierarchy = ConfigurationFunctions.hierarchical(allProperties, "edc.datasource.default");
```

Comment:
This is not clear to me, why in the test above the key for getting the property is only the part next to the root, instead here there's the need to report all the root key?
This hierarchy should countain only the elements in `edc.datasource.default` (user, password, driverClassName)

Reply:
We have two situations: 
1) there is an item between `root` and the elements. Here, that item would be `default`
2) there is no item between `root` and the elements

Now since both situations must return a map-of-maps, in the second case I need to introduce some sort of "artificial" root key, which is the entire prefix. 

Solving this without the "artificial root key" would require returning a tree structure instead of a map-of-maps which has 2 or 3 levels, depending on which situation we're in. Potentially even more.

Reply:
This behavior is a little tricky to understand at first sight, or better, it's not what I could expect from that method.
Would be better to return `Map<String, Object>` and leave to the client to check the type of the value, otherwise, the `Config` class would resolve all those issue nicely